src/chunk_append/chunk_append.c

```diff
@@ -281,3 +291,63 @@ contain_param_exec_walker(Node *node, void *context)
 
 	return expression_tree_walker(node, contain_param_exec_walker, context);
 }
+
+typedef struct
+{
+	Var *sort_var;
+	Var *ht_var;
+	Index sort_relid;
+	Index ht_relid;
+	Oid eq_opr;
+} find_equality_join_var_context;
+
+static bool
+find_equality_join_var_walker(Node *node, find_equality_join_var_context *context)
```

Comment:
should have a test with  a) > 2 hypertables involved in a  join, b) more than 2 hypertables and a non-hypertable involved in join. (not sure if this is possible since looks like we need the time partition column).